packages/interbit/src/tests/integrationTests/helpers.js

```diff
@@ -0,0 +1,41 @@
+const log = require('../../log')
+
+const waitForState = (chainInterface, predicate = () => {}, timeout = 2000) =>
+  new Promise((resolve, reject) => {
+    let unsubscribe = () => {}
+    unsubscribe = chainInterface.subscribe(() => {
+      const state = chainInterface.getState()
+      if (predicate(state)) {
+        unsubscribe()
+        resolve()
+      }
+    })
+
+    setTimeout(() => {
+      unsubscribe()
+      reject(new Error(`Waiting for block timed out after ${timeout}ms`))
+    }, timeout)
```

Comment:
The message is somewhat misleading. You are waiting for specific state, not a block.

Reply:
:+1: